packages/amplify-provider-awscloudformation/src/initializer.ts

```diff
@@ -165,13 +173,37 @@ function cloneCLIJSONForNewEnvironment(context) {
 export async function onInitSuccessful(context) {
   configurationManager.onInitSuccessful(context);
   if (context.exeInfo.isNewEnv) {
+    if (FeatureFlags.getBoolean('overrides.project')) {
+      await storeRootStackTemplate(context);
+    }
     context = await storeCurrentCloudBackend(context);
     await storeArtifactsForAmplifyService(context);
     await uploadHooksDirectory(context);
   }
   return context;
 }
 
+export const storeRootStackTemplate = async (context: $TSContext, template?: Template) => {
+  // generate template again as the folder structure was not created when root stack was initiaized
+  if (template === undefined) {
+    template = await transformRootStack(context);
+    await prePushCfnTemplateModifier(template);
+  }
+
+  // RootStack deployed to backend/awscloudformation/build
+  const projectRoot = pathManager.findProjectRoot();
+  const rootStackBackendBuildDir = pathManager.getRootStackDirPath(projectRoot);
```

Comment:
wouldn't you need to append the "build" path to the rootstckdir path?

Reply:
```
  getRootStackDirPath = (projectPath: string): string => {
    return this.constructPath(projectPath, [
      PathConstants.AmplifyDirName,
      PathConstants.BackendDirName,
      PathConstants.ProviderName,
      PathConstants.CfnStacksBuildDirName,
    ]);
  };
```

last is the build folder

Reply:
getRootStackDirPath shouldn't append the build dir or re-name the function to "getRootStackBuildDirPath"